config.go

```diff
@@ -49,3 +53,84 @@ func ReadConfig(path string) (*Config, error) {
 func WriteConfig(path string, cfg *Config) error {
 	return cuecfg.WriteFile(path, cfg)
 }
+
+// Formats for marshalling and unmarshalling a series of shell commands in a
+// devbox config.
+const (
+	// CmdArray formats shell commands as an array of of strings.
+	CmdArray CmdFormat = iota
+
+	// CmdString formats shell commands as a single string.
+	CmdString
+)
+
+// CmdFormat defines a way of formatting shell commands in a devbox config.
+type CmdFormat int
+
+func (c CmdFormat) String() string {
+	switch c {
+	case CmdArray:
+		return "array"
+	case CmdString:
+		return "string"
+	default:
+		return fmt.Sprintf("invalid (%d)", c)
+	}
+}
+
+// ConfigShellCmds marshals and unmarshals shell commands from a devbox config
+// as either a single string or an array of strings. It preserves the original
+// value such that:
+//
+//	data == marshal(unmarshal(data)))
+type ConfigShellCmds struct {
+	// MarshalAs determines how MarshalJSON encodes the shell commands.
+	// UnmarshalJSON will set MarshalAs automatically so that commands
+	// marshal back to their original format. The default zero-value
+	// formats them as an array.
+	//
+	MarshalAs CmdFormat
+	Cmds      []string
+}
+
+// MarshalJSON marshals shell commands according to s.MarshalAs. It marshals
+// commands to a string by joining s.Cmds with newlines.
+func (s ConfigShellCmds) MarshalJSON() ([]byte, error) {
+	switch s.MarshalAs {
+	case CmdArray:
+		return json.Marshal(s.Cmds)
+	case CmdString:
+		return json.Marshal(s.String())
+	default:
+		panic(fmt.Sprintf("invalid command format: %s", s.MarshalAs))
```

Comment:
I don't know whether we should do `panic` or return error. What's your take?

Reply:
I made it a panic because it's guaranteed to be a bug if it ever happens. Similar to panicking when a template fails to parse.